www/js/model.js

```diff
@@ -438,10 +472,17 @@ define([
 				var returnFlag = '';
 
 				var nameEntered = node.type && node.type == "equation" || node.variable;
-				// TODO : logic for value is still incomplete
-				// needs to incorporate values of function, accumulator and parameters
 				// as seen in Dragoon.
-				var valueEntered = node.type && node.type == "equation" || node.accumulator || node.value;
+				// variableType and value combined defines node completion
+				// node is complete in following cases
+				// 1. variableType Unknown and no value 
+				// 2. variableType dynamic and value is valid 
+				// 3. variableType parameter and value is valid
+				
+				var valueEntered = node.type && node.type == "equation" || (node.accumulator && node.value) 
+				|| (node.value == "" && node.variableType == "unknown") ||
+				(node.value && node.variableType == "parameter") ;
```

Comment:
we are checking node.value twice, would it be possible to change the if condition and say `(node.value && node.variableType != "unknown")`. This makes me think, the dynamic check becomes redundant. @jonwetzel this means I need to change this at many places for UI.